### DISCUSSION TITLE
一个歌手涉及到多个不同网站的时候如何决定namesapce？

### DISCUSSION THREAD

**Discussion by WonderfulSoap** [DIYgod/RSSHub#16837] [Q&A] [answered]
我打算为一个歌手(Ado)添加信息源，她的信息主要来自两个网站

环球音乐的歌手页 https://www.universal-music.co.jp/ado/news
歌手的粉丝俱乐部 https://ado-dokidokihimitsukichi-daigakuimo.com/articles/news

两者信息不重复。针对这种情况我应该按照[文档](https://docs.rsshub.app/joinus/new-rss/start-code#creating-namespace)这里所说，以下面形式分别独立创建两个 namesapce 吗？

```
lib/routes/universal-music
lib/routes/ado-dokidokihimitsukichi-daigakuimo
```

但这样有两个问题
- `universal-music`  这个namespace看起来是获取环球音乐下所有歌手信息的，但我实际上只打算提取特定歌手的信息（其他歌手页信息各有不同，我没精力维护）
- `ado-dokidokihimitsukichi-daigakuimo` 这个域名太长了作为namesapce的文件夹名真没问题吗
- 分成两个namespace后，在路由文档里会出现如下图这样的情况，明明是一个歌手，专门占用单独的两个大分类可以吗
![图片](https://github.com/user-attachments/assets/88c49512-be1f-417d-b1ee-016d0b68f278)

如果说namespace不按照域名区分，而是建立类似
```
lib/routes/ado-music
```
的形式的话似乎能解决问题，但这么做是允许的吗？

**Comment by TonyRL** [comment, marked as answer]
Use both `universal-music` and `ado-dokidokihimitsukichi-daigakuimo`

**Comment by WonderfulSoap** [comment]
Thanks
